File: apps/flutter/lib/core/tasks/caregiver_task.dart
enum TaskPriority { high, medium, low }

enum TaskStatus { pending, inProgress, completed }

class CaregiverTask {
  final String id;
  final String title;
  final String description;
  final TaskPriority priority;
  final TaskStatus status;
  final DateTime dueDate;
  final String? patientId;
  final String? patientName;
  final DateTime createdAt;
  final DateTime? completedAt;

  CaregiverTask({
    required this.id,
    required this.title,
    required this.description,
    required this.priority,
    required this.status,
    required this.dueDate,
    this.patientId,
    this.patientName,
    required this.createdAt,
    this.completedAt,
  });

  CaregiverTask copyWith({
    String? id,
    String? title,
    String? description,
    TaskPriority? priority,
    TaskStatus? status,
    DateTime? dueDate,
    String? patientId,
    String? patientName,
    DateTime? createdAt,
    DateTime? completedAt,
  }) {
    return CaregiverTask(
      id: id ?? this.id,
      title: title ?? this.title,
      description: description ?? this.description,
      priority: priority ?? this.priority,
      status: status ?? this.status,
      dueDate: dueDate ?? this.dueDate,
      patientId: patientId ?? this.patientId,
      patientName: patientName ?? this.patientName,
      createdAt: createdAt ?? this.createdAt,
      completedAt: completedAt ?? this.completedAt,
    );
  }

  bool get isOverdue => status != TaskStatus.completed && DateTime.now().isAfter(dueDate);
    bool get isDueToday {
    if (status == TaskStatus.completed) return false;
    if (isOverdue) return false;  // Exclude overdue tasks
    
    final now = DateTime.now();
    final today = DateTime(now.year, now.month, now.day);
    final taskDay = DateTime(dueDate.year, dueDate.month, dueDate.day);
    return today == taskDay;
    }
}

File: apps/react-native/src/models/CaregiverTask.js
export const TaskStatus = {
  pending: 'pending',
  inProgress: 'inProgress',
  completed: 'completed',
};

export default class CaregiverTask {
  constructor({ id, title, status = TaskStatus.pending, priority = 'medium', patient = '', dueDate = null, completedAt = null }) {
    this.id = String(id);
    this.title = title;
    this.status = status;
    this.priority = priority;
    this.patient = patient;
    this.dueDate = dueDate ? new Date(dueDate) : null;
    this.completedAt = completedAt ? new Date(completedAt) : null;
  }

  toJson() {
    return {
      id: this.id,
      title: this.title,
      status: this.status,
      priority: this.priority,
      patient: this.patient,
      dueDate: this.dueDate ? this.dueDate.toISOString() : null,
      completedAt: this.completedAt ? this.completedAt.toISOString() : null,
    };
  }

  static fromJson(raw) {
    return new CaregiverTask({
      id: raw.id,
      title: raw.title,
      status: raw.status,
      priority: raw.priority,
      patient: raw.patient,
      dueDate: raw.dueDate,
      completedAt: raw.completedAt,
    });
  }

  get isOverdue() {
    if (this.status === TaskStatus.completed) return false;
    if (!this.dueDate) return false;
    return Date.now() > this.dueDate.getTime();
  }

  get isDueToday() {
    if (!this.dueDate) return false;
    const d = new Date();
    return (
      d.getFullYear() === this.dueDate.getFullYear() &&
      d.getMonth() === this.dueDate.getMonth() &&
      d.getDate() === this.dueDate.getDate()
    );
  }
}
